crates/holochain/src/core/workflow/publish_dht_ops_workflow/unit_tests.rs

```diff
@@ -0,0 +1,295 @@
+use crate::core::queue_consumer::TriggerSender;
+use crate::core::queue_consumer::WorkComplete;
+use crate::core::workflow::publish_dht_ops_workflow::publish_dht_ops_workflow;
+use crate::prelude::AgentPubKeyFixturator;
+use crate::prelude::MockHolochainP2pDnaT;
+use crate::prelude::SignatureFixturator;
+use chrono::Utc;
+use fixt::prelude::*;
+use hdk::prelude::Action;
+use holo_hash::fixt::DnaHashFixturator;
+use holo_hash::AgentPubKey;
+use holo_hash::HasHash;
+use holochain_sqlite::db::DbKindAuthored;
+use holochain_sqlite::prelude::DatabaseResult;
+use holochain_sqlite::prelude::DbWrite;
+use holochain_state::prelude::set_receipts_complete;
+use holochain_state::prelude::StateMutationResult;
+use holochain_types::dht_op::DhtOp;
+use holochain_types::fixt::CreateFixturator;
+use holochain_types::prelude::DhtOpHash;
+use holochain_types::prelude::DhtOpHashed;
+use rusqlite::named_params;
+use std::sync::Arc;
+use std::time::{Duration, SystemTime};
+
+#[tokio::test(flavor = "multi_thread")]
+async fn no_ops_to_publish() {
+    holochain_trace::test_run().ok();
+
+    let test_db = holochain_state::test_utils::test_authored_db();
+    let vault = test_db.to_db();
+
+    let mut network = MockHolochainP2pDnaT::new();
+    network.expect_publish().never();
+
+    let (tx, rx) =
+        TriggerSender::new_with_loop(Duration::from_secs(5)..Duration::from_secs(30), true);
```

Comment:
Why does this sender have a loop?

Reply:
That doesn't work correctly as discussed above but that's another problem :) The workflow does correctly pick out ops on a timer and republish them

Reply:
Good stuff!